src/main/java/language/LanguageCommand.java

```diff
@@ -0,0 +1,383 @@
+                                                .collect(Collectors.toCollection(HashSet::new))
+                                        )
+                        ));
+            }
+
+            if (userLanguages.isEmpty()) {
+                return new EmbedBuilder()
+                        .setTitle("Failed to Calculate Linguistics Checks")
+                        .setDescription("Failed to find any input");
+            }
+            else {
+                Map<String, Try<List<Language>>> foundPaths = userLanguages.entrySet()
+                        .stream()
+                        .map(entry -> Pair.of(
+                                entry.getKey(),
+                                entry.getValue()
+                                        .flatMap(languages -> languageLogic.getPath(targetLanguageOpt.get(), languages))
+                        ))
+                        .collect(Collectors.toMap(Pair::getKey, Pair::getRight));
+
+                return new EmbedBuilder()
+                        .setTitle("Shortest Path to " + targetLanguageOpt.get())
+                        .setAuthor(user)
+                        .setDescription("The shortest path to " + targetLanguageOpt.get().getName() +
+                                " with this party is:\n" + joinPartyLinguisticsPaths(foundPaths)
+                        )
+                        .addField("Real Time Translation (Wits) / Grasp Idioms (Vision)", getGroupDifficultyString(foundPaths, 1))
+                        .addField("Translate Behind Conversation (Reason)", getGroupDifficultyString(foundPaths, 0))
+                        .setColor(RandomColor.getRandomColor());
+            }
+        }
+        else {
+            return new EmbedBuilder()
+                    .setTitle("Failed to Calculate Difficulty")
+                    .setDescription("No Target Language found for: " + option.getOptionStringValueByName("target").orElse("NONE"));
+        }
+    }
+
+    private String joinPartyLinguisticsPaths(Map<String, Try<List<Language>>> partyLanguageMap) {
+        return partyLanguageMap.entrySet().stream()
+                .map(stringListEntry -> String.format("%s: %s",
+                        stringListEntry.getKey(),
+                        stringListEntry.getValue()
+                                .fold(
+                                        Throwable::getMessage,
+                                        languages -> languages.stream()
+                                                .map(Language::getName)
+                                                .collect(Collectors.joining(" → "))
+                                )
+                ))
+                .collect(Collectors.joining("\n"));
+    }
+
+    private String getGroupDifficultyString(Map<String, Try<List<Language>>> partyLanguageMap, int modifier) {
+        return partyLanguageMap.entrySet().stream()
+                .map(stringListEntry -> String.format(
+                        "- %s: %s",
```

Comment:
Does this add the necessary roll number after the difficulty name?

Reply:
Not yet as I haven't added that logic to Tier